src/qt/sendcoinsdialog.cpp

```diff
@@ -461,6 +462,18 @@ void SendCoinsDialog::on_sendButton_clicked()
     m_current_transaction.reset();
 }
 
+void SendCoinsDialog::on_sendButton_clicked()
+{
+    try {
+        sendButtonClickedHelper();
+    } catch (const std::exception& e) {
+        Q_EMIT runawayException(QString::fromStdString(e.what()));
```

Comment:
Why emit a signal? No one is interested in this signal right? Could just call another slot in a generic catcher, like:
```cpp
template <typename F> void shieldException(F&& f) {
    try {
        f();
    } catch (...) {
        QMetaObject::invokeMethod(qApp, "messageException", Qt::QueuedConnection,
            Q_ARG(QString, exception_message), ...
        );
    }
}

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18897#issuecomment-625769660).